plugins/flatpak/gs-flatpak-transaction.c

```diff
@@ -62,6 +62,41 @@ gs_flatpak_transaction_set_no_deploy (FlatpakTransaction *transaction, gboolean
 	g_object_notify (G_OBJECT (self), "no-deploy");
 }
 
+/* Checks if a ref is a related ref to one of the installed ref.
+ * If yes, return the GsApp corresponding to the installed ref,
+ * NULL otherwise.
+ */
+static GsApp *
+get_installed_main_app_of_related_ref (FlatpakTransaction          *transaction,
+                                       FlatpakTransactionOperation *operation)
+{
+	FlatpakInstallation *installation = flatpak_transaction_get_installation (transaction);
+	const gchar *remote = flatpak_transaction_operation_get_remote (operation);
+	const gchar *op_ref = flatpak_transaction_operation_get_ref (operation);
+	GsFlatpakTransaction *self = GS_FLATPAK_TRANSACTION (transaction);
+	GList *keys;
+
+	if (g_str_has_prefix (op_ref, "app/"))
+		return NULL;
+
+	keys = g_hash_table_get_keys (self->refhash);
```

Comment:
Missed that during initial review, `keys` is leaking here.

Reply:
Addressed in latest iteration